link.go

```diff
@@ -102,13 +141,13 @@ func (l *link) attach(ctx context.Context, beforeAttach func(*frames.PerformAtta
 			l.muxClose(ctx, nil, nil, nil)
 		}()
 		return ctx.Err()
-	case <-l.session.done:
-		// session has terminated, no need to deallocate in this case
-		return l.session.doneErr
-	case fr = <-l.rx:
+	} else if err != nil {
+		return err
 	}
+
 	resp, ok := fr.(*frames.PerformAttach)
 	if !ok {
+		// TODO: seems like more cleanup is in order
```

Comment:
Worth filing an issue with some more details.

Reply:
This will be fixed in https://github.com/Azure/go-amqp/pull/237